frontend-cert/js-projects/quiz-game/user-stories.md

```diff
@@ -1,15 +1,10 @@
-1. You should create an object named `questions` with three keys, `science`, `geography`, and `literature`, each with the value of an empty array.
-1. You should create at least four `question#` variables, where `#` is replaced by a number starting from `1`.
-1. Each `question#` variable should be an object with the keys `question`, `choices`, and `answer`.
+1. You should create an array named `questions`.
+1. The `questions` array should contain at least five objects, each having the keys `category`, `question`, `choices`, and `answer`.
+1. The `category` key should have the value of a string representing a question category.
 1. The `question` key should have the value of a string representing a question.
 1. The `choices` key should have the value of an array containing three strings, which are alternative answers to the question.
 1. The `answer` key should have the value of a string, representing the correct answer to the question. Also, the value of `answer` should be included in the `choices` array.
-1. You should push each `question#` to the respective category inside the `questions` object.
-1. Each category of the `questions` object should have at least one element.
-1. You should declare a variable named `category` which has either the value of `"science"`, `"geography"`, or `"literature"`.
-1. You should remove the first element from the `questions[category]` array and store it inside a variable named `chosenQuestion`.
-1. You should log the value of the `question` key of `chosenQuestion` to the console.
-1. You should log the value of the `choices` key of `chosenQuestion` to the console.
-1. You should declare a variable named `answer`, which is a string.
-1. You should create an `if` statement that checks if `answer` is the correct answer and logs the string `The answer is correct!` to the console.
-1. You should write an `else` statement that logs the string `Wrong answer, the correct answer is: <correct-answer>`, where `<correct-answer>` is the value of the correct answer to the chosen question.
\ No newline at end of file
+1. You should have a function named `playQuiz` that selects a random question from the `questions` array.
+1. Your `playQuiz` function should use the `prompt` method to display a message and get user input.
```

Comment:
we ran into this with another workshop, where prompt doesn't work here because this is a node environment.
but since we are using functions now, you can just update the function to accept a parameter representing the user choice

Reply:
When the function is called the user doesn't know the question. How could they answer?

Reply:
In that case, you will need to refactor it to not accept user input all.
You can refactor this to have the input from the computer like this

```js
const questions = [
  {
    category: "science",
    question: "What is the chemical symbol for potassium?",
    choices: ["P", "K", "Pt"],
    answer: "K",
  },
  {
    category: "science",
    question: "What is the unit of electrical resistance?",
    choices: ["Ohm", "Coulomb", "Sievert"],
    answer: "Ohm",
  },
  {
    category: "geography",
    question: "What is the capital city of Australia?",
    choices: ["Sidney", "Canberra", "Wellington"],
    answer: "Canberra",
  },
  {
    category: "literature",
    question: 'Who wrote "1984"?',
    choices: ["Ray Bradbury", "Aldous Huxley", "George Orwell"],
    answer: "George Orwell",
  },
  {
    category: "sport",
    question: "How many players are on a standard volleyball team?",
    choices: ["6", "7", "12"],
    answer: "6",
  },
];

function getRandomQuestion() {
  const randomIndex = Math.floor(Math.random() * questions.length);
  return questions[randomIndex];
}

function getRandomComputerChoice(choices) {
  const randomIndex = Math.floor(Math.random() * choices.length);
  return choices[randomIndex];
}

function getResults(obj) {
  return computerChoice === obj.answer
    ? "The computer's choice is correct!"
    : `The computer's choice is wrong. The correct answer is: ${obj.answer}`;
}

const questionObj = getRandomQuestion();
// campers can either choose to use destructuring or dot/bracket notation to get these variables
const { question, choices } = questionObj;

console.log(question);
console.log(`Choices: ${choices}`);

const computerChoice = getRandomComputerChoice(choices);
console.log(`Computer chooses: ${computerChoice}`);

const results = getResults(questionObj);
console.log(results);

```